full_history_avd/src/history_tree/mod.rs

```diff
@@ -418,11 +428,11 @@ mod tests {
     impl MerkleTreeAVDParameters for MerkleTreeAVDTestParameters {
         const MAX_UPDATE_BATCH_SIZE: u64 = 4;
         const MAX_OPEN_ADDRESSING_PROBES: u8 = 2;
-        type MerkleTreeParameters = MerkleTreeTestParameters;
+        type Storer = MemStore<MerkleTreeTestParameters>;
     }
 
     type TestMerkleTreeAVD = MerkleTreeAVD<MerkleTreeAVDTestParameters>;
-    type TestAVDWithHistory = SingleStepAVDWithHistory<TestMerkleTreeAVD, MerkleTreeTestParameters>;
+    type TestAVDWithHistory = SingleStepAVDWithHistory<TestMerkleTreeAVD, dyn Storer<P = MerkleTreeTestParameters>>;
```

Comment:
@nirvantyagi is this right? Compiler wants dyn but I'm suspicious that it hasn't been needed elsewhere

Reply:
This is defining a concrete type, so you'll want to fill in the parameters with concrete types as well.

For example, here I think you will replace ``dyn Storer...`` with ``HTMemStore<MerkleTreeTestParameters>``. And then above in ``MerkleTreeAVDTestParameters``, I think you want the ``Storer`` type to be ``SMTMemStore<MerkleTreeTestParameters>``.

So this is related again to my comment above -- if I'm understanding things properly, it seems like you will need separate storer objects for history tree and sparse merkle tree. (The history tree storer should contain a sparse merkle tree storer within it). Does this make sense?